src/actions/receive-profile.js

```diff
@@ -85,9 +85,10 @@ export function viewProfile(
     pathInZipFile: string,
     implementationFilter: ImplementationFilter,
     transformStacks: TransformStacksPerThread,
+    geckoProfiler: $GeckoProfiler,
   |}> = {}
-): ThunkAction<void> {
-  return (dispatch, getState) => {
+): ThunkAction<Promise<void>> {
+  return async (dispatch, getState) => {
```

Comment:
This is a bit painful to make this async now. But this is probably the best way.